internal/service/sesv2/dedicated_ip_pool.go

```diff
@@ -0,0 +1,228 @@
+package sesv2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2"
+	"github.com/aws/aws-sdk-go-v2/service/sesv2/types"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func ResourceDedicatedIPPool() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceDedicatedIPPoolCreate,
+		ReadWithoutTimeout:   resourceDedicatedIPPoolRead,
+		UpdateWithoutTimeout: resourceDedicatedIPPoolUpdate,
+		DeleteWithoutTimeout: resourceDedicatedIPPoolDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"pool_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dedicated_ips": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"warmup_percentage": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"warmup_status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
```

Comment:
I am thinking about whether the IPs attribute (read-only) should be part of this resource.

Adding `aws_sesv2_dedicated_ip` resources will always cause a side effect on it. Example:
```terraform
resource "aws_sesv2_dedicated_ip_pool" "example" {
  pool_name = "my-pool"
}

# creating this resource will cause a side effect - will change the state of the pool
resource "aws_sesv2_dedicated_ip" "second" {
  destination_pool_name = aws_sesv2_dedicated_ip_pool.example.pool_name
  ip = "123"
}

# again, dedicated_ips of the pool will change
resource "aws_sesv2_dedicated_ip" "first" {
  destination_pool_name = aws_sesv2_dedicated_ip_pool.example.pool_name
  ip = "456"
}
```

Note: `aws_sesv2_dedicated_ip` is not implemented yet.

Reply:
Yeah, good point - I think you're right. 

My initial thinking was to provide that extra read-only info since we're getting it for free when checking for existence of the pool. However, the `aws_sesv2_dedicated_ip_pool` data source provides this, making it redundant on the resource anyway. I'll get this changed.